src/hotspot/share/classfile/classListParser.cpp

```diff
@@ -460,19 +460,40 @@ bool ClassListParser::is_matching_cp_entry(constantPoolHandle &pool, int cp_inde
   }
   return true;
 }
-
 void ClassListParser::resolve_indy(Symbol* class_name_symbol, TRAPS) {
+  ClassListParser::resolve_indy_impl(class_name_symbol, THREAD);
+  if (HAS_PENDING_EXCEPTION) {
+    ResourceMark rm(THREAD);
+    tty->print("resolve_indy for class %s has", class_name_symbol->as_C_string());
+    oop message = java_lang_Throwable::message(PENDING_EXCEPTION);
+    if (message != NULL) {
+      char* ex_msg = java_lang_String::as_utf8_string(message);
+      tty->print_cr(" exception pending '%s %s'",
+                    PENDING_EXCEPTION->klass()->external_name(), ex_msg);
+    } else {
+      tty->print_cr(" exception pending %s ",
+                    PENDING_EXCEPTION->klass()->external_name());
+    }
+    oop exception = THREAD->pending_exception();
+    if (exception->is_a(SystemDictionary::Error_klass())) {
+      CLEAR_PENDING_EXCEPTION;
```

Comment:
Just for my clarification, why do you clear the pending exception in the case of Error_klass()?  Please consider adding a comment explaining this.

Reply:
Hi Lois,

Thanks for your review. I've added some comment in `ClassListParser::resolve_indy` for the `Error_klass` case.

Reply:
I would have expected only LinkageErrors to be ignored as they are the only "expected" errors aren't they?

Reply:
LinkageError should work since BootstrapMethodError, NoSuchMethodError, etc. are subclasses.
I've changed the check to:
`if (exception->is_a(SystemDictionary::LinkageError_klass())) {`

Reply:
After offline discussion with Ioi, we've decided to use `log_warning(cds)` to output the exception message since log_warning does not require setting a log level. Also, the `exit(1)` is not needed, we can continue with the CDS dumping after printing the message and clearing the exception.